### PR TITLE
Add getFunctionSignature

### PR DESCRIPTION
**Pull request by volkyeth**
Adds the `getFunctionSignature` utility, as suggested on https://github.com/wagmi-dev/viem/discussions/764

Function signatures are used extensively on any Compound-based governance. Including this function makes it easier to migrate from ethers in those use cases.

---

### Focus of this PR:
This PR focuses on adding two new utility functions, `getFunctionSignature` and `getEventSignature`, to the codebase.

### Detailed summary:
- Added `getFunctionSignature` function to `src/utils/hash/getFunctionSignature.ts`.
- Added `getEventSignature` function to `src/utils/hash/getEventSignature.ts`.
- Updated `site/.vitepress/sidebar.ts` to include links to the new utility functions in the documentation sidebar.
- Updated `src/index.ts` to export the new utility functions.
- Added tests for `getFunctionSignature` in `src/utils/hash/getFunctionSignature.test.ts`.
- Updated documentation in `site/docs/utilities/getFunctionSignature.md` and `site/docs/utilities/getEventSignature.md` to include usage examples and descriptions of the new functions.

> The following files were skipped due to too many changes: `src/utils/hash/getFunctionSignature.test.ts`, `src/utils/hash/getEventSignature.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/utils/hash/getFunctionSelector.test.ts [hunk 1]
```diff
@@ -43,9 +43,9 @@ test('creates function signature from `AbiFunction`', () => {
       name: '_compound',
       type: 'function',
       inputs: [
-        { name: 'a', type: 'uint256' },
-        { name: 'b', type: 'uint256' },
-        { name: 'c', type: 'uint256' },
+        { name: '', type: 'uint256' },
```
Comment (volkyeth) [inline src/utils/hash/getFunctionSelector.test.ts:46]: this was a duplicate of the previous one. I imagine the intention was testing with named params

File: src/utils/hash/getFunctionSignature.test.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        { internalType: 'string', name: 'animationURI', type: 'string' },
+        { internalType: 'string', name: 'imageURI', type: 'string' },
+      ],
+      name: 'createEdition',
+      outputs: [{ internalType: 'address', name: '', type: 'address' }],
+      stateMutability: 'nonpayable',
+      type: 'function',
+    }),
+  ).toEqual(
+    'createEdition(string,string,uint64,uint16,address,address,(uint104,uint32,uint64,uint64,uint64,uint64,bytes32),string,string,string)',
```
Comment (volkyeth) [inline src/utils/hash/getFunctionSignature.test.ts:185]: h/t @ourzora 🫡

File: src/utils/hash/getFunctionSignature.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import type { AbiFunction } from 'abitype'
+
+import { formatAbiItem } from '../abi/formatAbiItem.js'
+import { extractFunctionName, extractFunctionParams } from '../index.js'
```
Comment (tmm) [inline src/utils/hash/getFunctionSignature.ts:4]: Let's import from the direct paths instead of the barrel import.

Comment (volkyeth) [reply src/utils/hash/getFunctionSignature.ts]: done in 2a9ba5e

Is this for the sake of consistency, or does it mess up tree-shaking or something?

Comment (jxom) [reply src/utils/hash/getFunctionSignature.ts]: It messes up tree-shaking.

Comment (volkyeth) [reply src/utils/hash/getFunctionSignature.ts]: Is there anything else I can do to help push this PR forward?

**Pull request merged**